File: app/src/main/java/com/gtm/pustak/notes/ui/dashboard/NotesDashBoardViewModel.java
package com.gtm.pustak.notes.ui.dashboard;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.gtm.pustak.notes.room.Notes;
import com.gtm.pustak.notes.room.NotesRepository;

import java.util.List;

public class NotesDashBoardViewModel extends AndroidViewModel {
    public LiveData<List<Notes>> mAllNotes;
    private NotesRepository repository;

    public NotesDashBoardViewModel(@NonNull Application application) {
        super(application);
        this.repository = new NotesRepository(application);
        mAllNotes = repository.getAllNotes();
    }


    public void deleteSelectedNote(Notes note) {

        repository.deleteNote(note);


    }

}

File: app/src/main/java/com/gtm/pustak/kotlin/ui/dashboard/NotesDashBoardViewModel.kt
package com.gtm.pustak.kotlin.ui.dashboard

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.gtm.pustak.notes.room.Notes
import com.gtm.pustak.notes.room.NotesRepository

class NotesDashBoardViewModel(application: Application) : AndroidViewModel(application) {

    var mAllNotes: LiveData<List<Notes>>
    var repository: NotesRepository? = null

    init {
        this.repository = NotesRepository(application)
        mAllNotes = repository!!.allNotes
    }


    fun deleteSelectedNote(note: Notes) {
        repository?.deleteNote(note)
    }
}